### PR TITLE
Introduce a move-session operation to the domain

### LINKED ISSUE #26: create domain/move-session function
A rough sketch of the `spacy.domain/move-session` function:

```clojure
(defn move-session [state _current-user {:keys [id room time]}]
  {:post [(s/valid? ::outcome %)]}
  {::facts [#_(could use ::session-scheduled to begin with, or create ::session-moved identical to ::session-scheduled )]
   ::event (-> state
               #_(update function which moves session from one slot in the schedule to another ))}))
```

`move-session` should probably produce an error if:
* the session is not in `::schedule`
* the target slot is not open

Tests probably wouldn't be a bad idea.

### PR DESCRIPTION
**Pull request by jstepien**
The moved session map is created eagerly, even if moving is impossible. I'm assuming that most real-life invocations will lead to a valid move so  it shouldn't be a problem.

Fixes #26.

I'm also thinking about a patch that aliases spacy.domain as domain instead of sut. Then I'd shorten all those domain keywords, e.g. `:spacy.domain/error` → `::domain/error`. What do you think?

### CODE REVIEW

**Comment by joyheron** [comment]
I'd prefer `::domain/error` to `:spacy.domain/error` 👍

File: src/spacy/domain.clj [hunk 1]
```diff
@@ -172,6 +178,23 @@
...
+        is-the-sponsor? (= current-user (::sponsor session))
+        can-move? (and session
+                       is-the-sponsor?
+                       (is-open-slot? state room time))
+        moved (assoc session ::room room ::time time)]
+    (if can-move?
+      {::event (-> state
+                   (update ::schedule (partial replace {session moved})))
+       ::facts [(assoc moved
+                       ::fact ::session-moved
```
Comment (joyheron) [inline src/spacy/domain.clj:193]: Do we want to add `::session-moved` to the set in `::facts` in L54

Comment (jstepien) [reply src/spacy/domain.clj:193]: Oh, good catch. I forgot to add it and yet the postcondition above passes.

I'm wondering if it's a sign that our specs are too loose or that we're using them incorrectly.

I'll look into it!

**Review by joyheron — APPROVED** [review]

File: src/spacy/domain.clj (hunk 1 above)
Comment (jstepien) [reply src/spacy/domain.clj:193]: → #44

**Pull request merged**